src/katgpucbf/fgpu/engine.py

```diff
@@ -901,27 +902,42 @@ async def _chunk_send_and_cleanup(
                 chunk.cleanup()
                 chunk.cleanup = None  # Potentially helps break reference cycles
 
-    def _update_dig_power_sensors(self, dig_total_power: accel.HostArray | None, out_item: OutQueueItem) -> None:
+    def _update_dig_power_sensors(
+        self,
+        dig_total_power_accums: list[Accum],
+        dig_total_power: accel.HostArray,
+        out_item: OutQueueItem,
+    ) -> None:
         """Update digitiser power sensors.
 
-        Helper function for keeping the complexity of :meth:`run_transmit` down to manageable levels.
+        Parameters
+        ----------
+        dig_total_power_accums
+            Accumulators tracking long-term digitiser total power (one per polarisation)
+        dig_total_power
+            The total power per polarisation in `out_item`
+        out_item
+            The current :class:`OutQueueItem`
         """
-        if dig_total_power is not None:
-            update_timestamp = self.engine.time_converter.adc_to_unix(out_item.next_timestamp)
-            if np.all(out_item.present):
-                for pol, trg in enumerate(dig_total_power):
-                    total_power = float(trg)
-                    avg_power = total_power / (out_item.n_spectra * self.output.spectra_samples)
+        all_present = np.all(out_item.present)
+        for pol, (accum, trg) in enumerate(zip(dig_total_power_accums, dig_total_power)):
+            power: int | None = int(trg)
+            if not all_present:
+                power = None
+            if measurement := accum.add(out_item.timestamp, out_item.next_timestamp, power):
```

Comment:
Walrus operator warming the cockles of my heart...

Perhaps I have misunderstood, but I thought that if `measurement` is `None` then you also want to update the sensor with Failure but it doesn't look as though it happens. Currently if `measurement` isn't none but `measurement.value` is none, there's a Failure update.

Reply:
`measurement is None` means "not enough time has passed since the last measurement." `measure.value is None` means "enough time has passed and we missed some data since the last measurement".